test/whl-scale-tests/deploy-log-generators.ps1

```diff
@@ -0,0 +1,120 @@
+<#
+.SYNOPSIS
+Deploy log generators
+.DESCRIPTION
+Deploys one or more log generators for scale testing. Can specify CrashDumps, ETW, EventLogs, and/or TextLogs. If none specified all will be deployed
+.PARAMETER CrashDumps
+Deploy the crash dumps generator
+.PARAMETER ETW
+Deploy the ETW generator
+.PARAMETER EventLogs
+Deploy the Event Logs generator
+.PARAMETER TextLogs
+Deploy the Text Logs generator
+.EXAMPLE
+deploy-log-generators.ps1
+
+Deploy all Generators.
+.EXAMPLE
+deploy-log-generators.ps1 -TextLogs
+
+Deploy just the text logs generator
+.EXAMPLE
+deploy-log-generators.ps1 -ETW -EventLogs
+
+Deploy the ETW and Event Log generators
+#>
+param(
+  [Parameter()][switch]$CrashDumps,
+  [Parameter()][switch]$ETW,
+  [Parameter()][switch]$EventLogs,
+  [Parameter()][switch]$TextLogs
+)
+
+. "$PSScriptRoot\common.ps1"
+
+$all = !$CrashDumps -and !$ETW -and !$EventLogs -and !$TextLogs
+
+<#
+.SYNOPSIS
+Deploy a log generation DaemonSet
+.DESCRIPTION
+Build and deploy a DaemonSet to generate logs
+.PARAMETER imageTag
+Full tag to use for the container image in the format <uri>/<tag>:<version>
+.PARAMETER name
+Name to give the DaemonSet
+.PARAMETER namespace
+Kubernetes namespace where the DaemonSet should be deployed
+.PARAMETER nodeSelector
+String to match for selecting nodes where the DaemonSet should be deployed
+.PARAMETER buildPath
+Docker Context to use for build
+.PARAMETER dockerfilePath
+Path to dockerfile. Default is $buildPath/Dockerfile
+.PARAMETER preBuild
+Setup script block to run before building docker image
+.EXAMPLE
+Deploy-LogGenerator exampleacr.azurecr.io/generatelogs:latest LogGenerator log-generation whl-logs
+#>
+function buildAndDeploy {
```

Comment:
Is it best practice to use camel case with local functions?

Reply:
I think Pascal case is best practice. Updated